modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -321,51 +321,49 @@ def broadcast_apply(cls, axis, apply_func, left, right, other_name="r"):
             Axis to apply and broadcast over.
         apply_func : callable
             Function to apply.
-        left : NumPy 2D array
-            Left partitions.
-        right : NumPy 2D array
-            Right partitions.
+        left : np.ndarray
+            NumPy array of left partitions.
+        right : np.ndarray
+            NumPy array of right partitions.
         other_name : str, default: "r"
             Name of key-value argument for `apply_func` that
             is used to pass `right` to `apply_func`.
 
         Returns
         -------
-        NumPy array
-            An of partition objects.
+        np.ndarray
+            NumPy array of result partition objects.
 
         Notes
         -----
         This will often be overridden by implementations. It materializes the
         entire partitions of the right and applies them to the left through `apply`.
         """
-        [obj.drain_call_queue() for row in right for obj in row]
-        new_right = np.empty(shape=right.shape[axis], dtype=object)
 
-        if axis:
-            right = right.T
+        def map_func(df, *others):
+            other = pandas.concat(others, axis=axis ^ 1)
+            return apply_func(df, **{other_name: other})
 
-        for i in range(len(right)):
-            new_right[i] = pandas.concat(
-                [right[i][j].get() for j in range(len(right[i]))], axis=axis ^ 1
-            )
-        right = new_right.T if axis else new_right
-
-        new_partitions = np.array(
+        [obj.drain_call_queue() for row in right for obj in row]
```

Comment:
The overwriden methods weren't draining the call queue for every partition to broadcast. This draining can cost a lot in case of delayed tasks being assigned to partitions.

Reply:
It seems to me that we should call this function, otherwise we will work with non-actual data.  Working with internal features bypassing the partition class itself can lead to loss of operations.

However, I figured out how it worked before. `drain_call_queue()` function is called in the `axis_partition` constructor.

Reply:
@devin-petersohn, is this a solid rule now, that nothing except `Partition`/`AxisPartition` classes mustn't drain partition's call queue, and so we can safely remove this draining because it's now partition classes responsibility to operate with the queue?

Reply:
`Partition` and `AxisPartition` classes drain a call queue only for self object for now, but not for other/right dataframe partitions. However, we likely want to drain the call queues for self and right/other objects in a remote call.

Reply:
> @devin-petersohn, is this a solid rule now, that nothing except `Partition`/`AxisPartition` classes mustn't drain partition's call queue, and so we can safely remove this draining because it's now partition classes responsibility to operate with the queue?

I think at the moment there is no such rule, let's leave this "extra" check here for now, since it does not take much time.